File: Python/algorithm/src/insertion_sort.py
from src.log import log


def insertion_sort(values):
    log.info(values)
    i = 1
    for i in range(len(values)):
        j = i - 1
        while j >= 0:
            if values[j] > values[j + 1]:
                key = values[j + 1]
                values[j + 1] = values[j]
                values[j] = key
                j -= 1
            else:
                break
        i += 1
    log.info(values)

File: Go/algorithm/src/insertion_sort.go
package src

func InsertionSort(values []int) (ret []int, err error) {
	//start := time.Now()
	for i := 1; i < len(values); i++ {
		for j := i - 1; j >= 0; j-- {
			if values[j] > values[j+1] {
				key := values[j]
				values[j] = values[j+1]
				values[j+1] = key
			} else {
				break
			}
		}
	}
	var logger = GetLogger()
	logger.Info("insertion sort cost:")
	return values, nil
}
